module/VuFind/src/VuFind/Controller/ShortlinkController.php

```diff
@@ -50,7 +76,19 @@ public function redirectAction()
         if ($id = $this->params('id')) {
             $resolver = $this->serviceLocator->get(UrlShortenerInterface::class);
             if ($url = $resolver->resolve($id)) {
-                return $this->redirect()->toUrl($url);
+                switch ($this->redirectMethod) {
+                case 'html':
+                    $view = $this->createViewModel();
+                    $view->redirectTarget = $url;
+                    $view->redirectDelay = 3;
```

Comment:
Maybe it's worth making the redirect delay a class property so it can be made configurable or overridden in future if needed.